src/Fabulous.Avalonia/Widgets/Controls/TextBox.fs

```diff
@@ -0,0 +1,166 @@
+
+open System.Runtime.CompilerServices
+open Avalonia.Controls
+open Avalonia.Media
+open Fabulous
+open Avalonia.Layout
+
+type IFabTextBox =
+    inherit IFabTemplatedControl
+
+module TextBox =
+    let WidgetKey = Widgets.register<TextBox> ()
+
+    let Text = Attributes.defineAvaloniaPropertyWithEquality TextBox.TextProperty
+
+    let TextAlignment = Attributes.defineAvaloniaPropertyWithEquality TextBox.TextAlignmentProperty
+
+    let TextWrapping = Attributes.defineAvaloniaPropertyWithEquality TextBox.TextWrappingProperty
+
+    let HorizontalContentAlignment = Attributes.defineAvaloniaPropertyWithEquality TextBox.HorizontalContentAlignmentProperty
+
+    let VerticalContentAlignment = Attributes.defineAvaloniaPropertyWithEquality TextBox.VerticalContentAlignmentProperty
+
+    let AcceptsReturn = Attributes.defineAvaloniaPropertyWithEquality TextBox.AcceptsReturnProperty
+    
+    let AcceptsTab = Attributes.defineAvaloniaPropertyWithEquality TextBox.AcceptsTabProperty
+    
+    let IsReadOnly = Attributes.defineAvaloniaPropertyWithEquality TextBox.IsReadOnlyProperty
+
+    let IsUndoEnabled = Attributes.defineAvaloniaPropertyWithEquality TextBox.IsUndoEnabledProperty
+
+    let LetterSpacing = Attributes.defineAvaloniaPropertyWithEquality TextBox.LetterSpacingProperty
+  
+    let LineHeight = Attributes.defineAvaloniaPropertyWithEquality TextBox.LineHeightProperty
+  
+    let MaxLength = Attributes.defineAvaloniaPropertyWithEquality TextBox.MaxLengthProperty
+  
+    let MaxLines = Attributes.defineAvaloniaPropertyWithEquality TextBox.MaxLinesProperty
+    
+    let PasswordChar = Attributes.defineAvaloniaPropertyWithEquality TextBox.PasswordCharProperty
+
+    let RevealPassword = Attributes.defineAvaloniaPropertyWithEquality TextBox.RevealPasswordProperty    
+
+    let Watermark = Attributes.defineAvaloniaPropertyWithEquality TextBox.WatermarkProperty
+
+    let UseFloatingWatermark = Attributes.defineAvaloniaPropertyWithEquality TextBox.UseFloatingWatermarkProperty
+
+    // Read-only Properties
+    let CanCopy = Attributes.defineAvaloniaPropertyWithEquality TextBox.CanCopyProperty
+    
+    let CanCut = Attributes.defineAvaloniaPropertyWithEquality TextBox.CanCutProperty
+    
+    let CanPaste = Attributes.defineAvaloniaPropertyWithEquality TextBox.CanPasteProperty
+  
+    let CaretIndex = Attributes.defineAvaloniaPropertyWithEquality TextBox.CaretIndexProperty    
+
+    let NewLine = Attributes.defineAvaloniaPropertyWithEquality TextBox.NewLineProperty
+
+    // Property Widgets
+    let CaretBrush = Attributes.defineAvaloniaPropertyWidget TextBox.CaretBrushProperty
+
+    let SelectionBrush = Attributes.defineAvaloniaPropertyWidget TextBox.SelectionBrushProperty
+   
+    let SelectionForegroundBrush = Attributes.defineAvaloniaPropertyWidget TextBox.SelectionForegroundBrushProperty
+
+    // Events
+    let ValueChanged = Attributes.defineAvaloniaPropertyWithChangedEvent' "TextBox_ValueChanged" TextBox.TextProperty
+
```

Comment:
Can we also add event wrappers for :
- CopyingToClipboardEvent
- PastingFromClipboardEvent

You can use **Attributes.defineEvent**
ie:
```fsharp
let CopyingToClipboardEvent =
        Attributes.defineEvent<RoutedEventArgs> "TextBox_CopyingToClipboardEvent" (fun target ->
            (target :?> TextBox).CopyingToClipboard)
```